controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -212,14 +212,16 @@ func (r *BareMetalHostReconciler) Reconcile(request ctrl.Request) (result ctrl.R
 		return ctrl.Result{}, errors.Wrap(err, "failed to create provisioner")
 	}
 
-	ready, err := prov.IsReady()
-
-	if err != nil {
-		return ctrl.Result{}, errors.Wrap(err, "failed to check services availability")
-	}
-	if !ready {
-		reqLogger.Info("provisioner is not ready", "RequeueAfter:", provisionerNotReadyRetryDelay)
-		return ctrl.Result{Requeue: true, RequeueAfter: provisionerNotReadyRetryDelay}, nil
+	// if we can't check the ready state assume it is up.
+	if host.HasBMCDetails() {
```

Comment:
This one it's not clear to me. The status of the provisioner is completely independent by that, ie Ironic could be down. The `IsReady()` check was introduced also to skip performing any action in the state machine if the provisioner was not available, in this way we're going to re-introduce the old behavior

Reply:
We don't actually do anything in the Unmanaged state, so it doesn't matter.

Reply:
Can we end up then in a situation where the state is not Unmanaged _and_ bmc details are missing? Otherwise, could make sense to add this constraint to the check? From the PR description it's not fully clear the context, or if there are other restrinctions

Reply:
So the reason I did this was, the ready check also checks dependancies (ironic which does not exist in assisted installer and thus will always return an error).

Reply:
Also in the normal (non-AI case) the BMC details should be there, but ironic might be down (it will then correctly fail). HasBMCDetails() is just checking the configuration.
I don't want to add a check for state!=unmanaged and HasBMCDetails() as to get to unmanaged it needs to go through the state machine.. it starts off in state=""

Reply:
Thanks, let's continue the discussion upstream